pkg/policy/groups/helpers.go

```diff
@@ -123,24 +169,74 @@ func updateDerivativeStatus(cnp *cilium_v2.CiliumNetworkPolicy, derivativeName s
 		status.OK = true
 	}
 
+	if clusterScoped {
+		return updateDerivativeCNPStatus(cnp, status, derivativeName)
+	}
+
+	return updateDerivativeCCNPStatus(cnp, status, derivativeName)
+}
+
+func updateDerivativeCNPStatus(cnp *cilium_v2.CiliumNetworkPolicy, status cilium_v2.CiliumNetworkPolicyNodeStatus,
+	derivativeName string) error {
 	// This CNP can be modified by cilium agent or operator. To be able to push
 	// the status correctly fetch the last version to avoid updates issues.
-	k8sCNPStatus, clientErr := k8s.CiliumClient().CiliumV2().
-		CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).
+	k8sCNP, clientErr := k8s.CiliumClient().CiliumV2().CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).
 		Get(context.TODO(), cnp.ObjectMeta.Name, v1.GetOptions{})
+
 	if clientErr != nil {
-		return fmt.Errorf("Cannot get Kubernetes policy: %s", clientErr)
+		return fmt.Errorf("cannot get Kubernetes policy: %s", clientErr)
```

Comment:
Nit: in the places that you've modified the error return, change the format string to `%v` for errors, instead of `%s`.